File: 351002/Valinski/lab5_6/RedisTraining/publisher/src/Publisher.Domain/Models/Topic.cs
namespace Publisher.Domain.Models;

public class Topic
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }

    public User? User { get; set; }
    public List<Label> Labels { get; set; } = new List<Label>();
}

File: 351001/Pauliuchenka/lib/domain/src/entities/topic.rs
use chrono::{DateTime, Utc};
use sqlx::prelude::FromRow;
use validator::Validate;

use crate::entities::IDType;

/// Topic entity
#[derive(Debug, Clone, Validate, FromRow)]
pub struct Topic {
    pub id: IDType,
    pub creator_id: IDType,
    #[validate(length(min = 2, max = 64))]
    pub title: String,
    #[validate(length(min = 4, max = 2048))]
    pub content: String,
    /// ISO8601
    pub created: DateTime<Utc>,
    /// ISO8601
    pub modified: DateTime<Utc>,
}
